rust/lance-index/src/vector/pq/utils.rs

```diff
@@ -13,8 +15,18 @@ use lance_linalg::MatrixView;
 pub(super) fn divide_to_subvectors<T: ArrowFloatType>(
     data: &MatrixView<T>,
     m: usize,
-) -> Vec<Arc<T::ArrayType>> {
+) -> Result<Vec<Arc<T::ArrayType>>> {
     assert!(!data.num_rows() > 0);
+    if data.num_columns() % m != 0 {
+        return Err(Error::invalid_input(
+            format!(
+                "num_sub_vectors must divide vector dimension {}, but got {}",
```

Comment:
If possible, I might refer the user in a section to the docs that explains this in more detail. I'd imagine that some users just run the index command expecting it to work, and if it doesn't they probably need more than this message to debug it.

Reply:
Unfortunately, we don't yet have a docs page that describes this well. We should add that, but this seems like a definite improvement over the current panics.